### PR TITLE
Refactor/hydro on dom

### PR DESCRIPTION
**Pull request by leighlars**
### Type of change made:
- [X] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Styling- no new features
- [X] Refactoring
- [ ] Planning

### Detailed Description
User can now view the overall community's daily average water consumption. I also refactored hydro repo. The user can average their overall average of daily water consumption, average a week's worth of  water consumption, and get a list of all water consumption per day over a week. It is also displayed on the DOM Hydro card. 

### Why is this change required? What problem does it solve?
These issues were part of the spec. 

### Were there any challenges that arose while implementing this feature? If so, how were the addressed?
I was unable to clear the input field after the user submits water info for the day. 

### Files modified:
- [ ] index.html
- [ ] .scss // file name(s):
- [X] classes // file name(s): Deleted Hydration class. Modified HydroRepo
- [X ] test // files name(s): HydroRepo test
- [X] scripts.js
- [X] Other files: DOM updates

### CODE REVIEW

File: src/HydrationRepository.js [hunk 1]
```diff
@@ -1,47 +1,54 @@
 class HydrationRepository {
   constructor(today) {
     this.individualEntryRecords = [];
-    // this.todaysWaterData = this.findTodaysTotalWater(today);
     this.averageOuncesAllTime = this.getAverageOuncesOverall();
   }
   
+  getOuncesByDay(date) {
+    let dayFound = this.individualEntryRecords.find(entry => entry.date === date);
+    return dayFound ? dayFound.numOunces : 0;
```
Comment (NakiNorton) [inline src/HydrationRepository.js:9]: Great use of a ternary here. Very clean.

File: test/HydrationRepository-test.js [hunk 2]
```diff
@@ -3,15 +3,18 @@ import HydrationRepository from '../src/HydrationRepository';
...
     hydrate4 = { userID: 1, date: "07/18/2020", numOunces: 20 }
-    mockRawData = [hydrate1, hydrate2, hydrate3, hydrate4]
-    todayDate = "07/20/2020"
-    mockHydroRepo = new HydrationRepository(todayDate)
+    hydrate5 = { userID: 1, date: "07/20/2020", numOunces: 30 }
+    hydrate6 = { userID: 1, date: "07/18/2020", numOunces: 20 }
+    hydrate7 = { userID: 1, date: "07/18/2020", numOunces: 20 }
+    mockRawData = [hydrate1, hydrate2, hydrate3, hydrate4, hydrate5, hydrate6, hydrate7]
+    todaysDate = "07/21/2020"
+    mockHydroRepo = new HydrationRepository(todaysDate)
```
Comment (NakiNorton) [inline test/HydrationRepository-test.js:17]: Nice work setting up a mock data set.

**Comment by NakiNorton** [comment]
Thanks for all your work on the hydration class! This is looking great. I'm running into the same issue with clearing input fields after submission, it won't work when I try to do it from domUpdates, even though that's where this should be done. It's something we can take a look at once we're finished getting these last major pieces of functionality working.

File: src/HydrationRepository.js [hunk 3]
```diff
@@ -1,47 +1,54 @@
...
+    let weekAverageOunces = (numOuncesTotal / 7).toFixed(0);
+    return Number(weekAverageOunces);
   }
 
-  getWeeksDailyOunces() {
-    let week = this.individualEntryRecords.slice(-7)
-    return week.map(day => {
-      return day.numOunces;
-    });
+  addHydroInfo(input) {
```
Comment (Leasw144) [inline src/HydrationRepository.js:43]: I still think info should be replaced with stats for clarity, thoughts or would this be too much to bother with?

File: src/domUpdates.js [hunk 4]
```diff
@@ -137,20 +137,17 @@ const domUpdates = {
...
-  //   // input.value = "";
-  // },
-  
-  
-  // document.querySelector("#hydration-friend-ounces-today").innerText = `${this.userRepository.calculateAverageDailyWater(this.todaysDate)}`;
 
+  hydrationCardDisplay() {
+    document.getElementById("hydration-user-ounces-today").innerText = `${this.currentUser.hydrationInfo.getOuncesByDay(this.todaysDate)}`;
+    document.querySelector(".hydration-weekly-avg").innerText = `You averaged ${this.currentUser.hydrationInfo.getWeekAvgOunces()} ounces this week!`;
+    let dailyOz = document.querySelectorAll('.daily-oz');
```
Comment (Leasw144) [inline src/domUpdates.js:144]: Should the variables be declared before the query selectors or do you think this doesn't matter. More of a stylistic comment than anything else.

**Comment by Leasw144** [comment]
Looks great, thanks for doing this.

**Pull request merged**